src/video_core/gpu.cpp

```diff
@@ -66,6 +67,50 @@ const DmaPusher& GPU::DmaPusher() const {
     return *dma_pusher;
 }
 
+void GPU::IncrementSyncPoint(const u32 syncpoint_id) {
+    syncpoints[syncpoint_id]++;
+    sync_mutex.lock();
+    if (!syncpt_interrupts[syncpoint_id].empty()) {
+        u32 value = syncpoints[syncpoint_id].load();
+        auto it = syncpt_interrupts[syncpoint_id].begin();
+        while (it != syncpt_interrupts[syncpoint_id].end()) {
+            if (value >= *it) {
+                TriggerCpuInterrupt(syncpoint_id, *it);
+                it = syncpt_interrupts[syncpoint_id].erase(it);
+                continue;
+            }
+            it++;
+        }
+    }
+    sync_mutex.unlock();
+}
+
+u32 GPU::GetSyncpointValue(const u32 syncpoint_id) const {
+    return syncpoints[syncpoint_id].load();
+}
+
+void GPU::RegisterSyncptInterrupt(const u32 syncpoint_id, const u32 value) {
+    for (u32 in_value : syncpt_interrupts[syncpoint_id]) {
+        if (in_value == value)
+            return;
+    }
+    syncpt_interrupts[syncpoint_id].emplace_back(value);
+}
+
+bool GPU::CancelSyncptInterrupt(const u32 syncpoint_id, const u32 value) {
+    sync_mutex.lock();
+    auto it = syncpt_interrupts[syncpoint_id].begin();
+    while (it != syncpt_interrupts[syncpoint_id].end()) {
```

Comment:
This can use std::find_if

```suggestion
    auto& interrupt = syncpt_interrupts[syncpoint_id];
    const auto iter = std::find_if(interrupt.begin(), interrupt.end(), [value](u32 interrupt_value) {
        return value == interrupt_value;
    });
    
    if (iter == interrupts.end()) {
        return false;
    }

    interrupts.erase(iter);
    return true;
```